### DISCUSSION TITLE
Feature Request: Masked e-mail integration via Fastmail

### DISCUSSION THREAD

**Discussion by goose-ws** [dani-garcia/vaultwarden#2790] [Ideas]
As of September, BitWarden can interface with the FastMail API to generate masked/forwarded e-mail addresses when creating accounts. In short, this allows you to create a unique e-mail for each login through Fastmail, which all end up in the same inbox.

Relevant blog posts:
https://bitwarden.com/blog/use-bitwarden-to-generate-email-aliases-with-fastmail/
https://fastmail.blog/company/masked-email-now-in-more-places-with-bitwarden-integration/

Any chance this could be implemented in VaultWarden as well?

**Comment by BlackDex** [comment]
This is already implemented within the `testing` tagged images which are based upon the `main` branch.

**Reply by goose-ws** [reply]
Oh nice! I tried to search for any previous submission for the feature request and didn't find any, so I didn't think it was. But that's great! Thanks!

**Reply by cospeedster** [reply]
Is it now GA?

**Reply by BlackDex** [reply]
For more then 2 years already

**Reply by s3rverro0m** [reply]
I finally decided to switch to Fastmail and it's been amazing! I'm seeing a weird issue with Bitwarden on iOS that I'm not seeing on the Web or macOS. Has anyone seen this?
![image](https://github.com/user-attachments/assets/d26cc2bf-0ad9-4afa-9911-967a59511be8)

I looked at server logs and they don't show anything useful.

**Reply by goose-ws** [reply]
If your problem is with the iOS BitWarden app, and not the VaultWarden web server/panel, you may be better off posting in the [BitWarden Clients GitHub](https://github.com/bitwarden/clients).

**Reply by s3rverro0m** [reply]
Very true! I'll give that a shot as well.
